examples/src/main/scala/zio/config/examples/ProgramExample.scala

```diff
@@ -19,77 +20,108 @@ object ProgramExample extends App {
   case class Live(config: Config.Service[ProgramConfig], spark: SparkEnv.Service)
       extends SparkEnv
       with Config[ProgramConfig]
+      with Blocking.Live
 
   override def run(args: List[String]): ZIO[Environment, Nothing, Int] = {
-    val pgm = for {
-      confService <- Config.fromEnv(programConfig)
-      session     <- ZIO.effect("sparkSession")
-      _           <- Application.execute.provide(Live(confService.config, new SparkEnv.Live(session)))
-    } yield ()
+    val pgm =
+      for {
+        configEnv <- Config.fromEnv(programConfig)
+        sparkEnv  <- SparkEnv.local("some-app")
+        _         <- Application.execute.provide(Live(configEnv.config, sparkEnv.spark))
+      } yield ()
 
-    ZIO.accessM[Environment](env => {
+    ZIO.accessM[Environment] { env =>
       pgm.foldM(
         fail => env.console.putStrLn(s"failed $fail") *> ZIO.succeed(1),
         _ => env.console.putStrLn(s"succeeded") *> ZIO.succeed(0)
       )
-    })
+    }
   }
 }
 
-// spark service
+//// stubs for the real Spark
+
+trait SparkSession {
+  def sql(value: String): DataFrame =
+    new DataFrame {}
+
+  def version: String =
+    "someVersion"
+}
+
+trait DataFrame
+
+////
+
 trait SparkEnv {
   def spark: SparkEnv.Service
 }
 
 object SparkEnv {
-  type SparkSession = String
-
   trait Service {
-    def getSpark: UIO[SparkSession]
+    def sparkEnv: UIO[SparkSession]
   }
 
-  class Live(r: SparkSession) extends Service {
-    def getSpark: UIO[SparkSession] = ZIO.succeed(r)
-  }
+  def make(session: => SparkSession): ZIO[Blocking, Throwable, SparkEnv] =
+    zio.blocking
+      .effectBlocking(session)
+      .map(
+        sparkSession =>
+          new SparkEnv {
+            override def spark: Service =
+              new Service {
+                override def sparkEnv: UIO[SparkSession] =
+                  ZIO.succeed(sparkSession)
+              }
+          }
+      )
+
+  def local(name: String): ZIO[Blocking, Throwable, SparkEnv] =
+    make {
+      // SparkSession.builder().appName(name).enableHiveSupport().getOrCreate()
```

Comment:
Add a comment,
An example , that it is an example of real world spark initiation.